Fix the Advanced Settings Aren't Searchable, #4512

low-batt opened the pull request:
This commit will:
- Add `SectionTitleTop` hidden label to `Advanced` tab
- Add `SectionTitleLogging` hidden label to `Advanced` tab
- Add `FunctionalButtonShowLog` identifier to `Show log viewer` button on  `Advanced` tab
- Add `FunctionalButtonLogDirectory`  identifier to `Open log directory` button on  `Advanced` tab
- Add `FunctionalButtonShowConfig` identifier to `Show the config file in Finder` button on `Key Bindings` tab
- Add `FunctionalButtonImportConfig` identifier to `Import an existing config file` button on `Key Bindings` tab
- Add `FunctionalButtonDefaultApp` identifier to `Set IINA as the Default Application…` button on `Utilities` tab
- Add `formSearchTerm` method to `PreferenceWindowController` that removes extraneous characters from labels and titles
- Correct the spelling of available on `General` tab
- Change `Use media keys` to be  `Use system media control` to match UI
- Add a `logLabelDict` method to `PreferenceWindowController` that logs the search terms found by scanning NIBs

- [x] I have read [CONTRIBUTING.md](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md)
- [x] This implements/fixes issue #4512.

---

**Description:**

low-batt committed:
Fix the Advanced Settings Aren't Searchable, #4512

This commit will:
- Add SectionTitleTop hidden label to Advanced tab
- Add SectionTitleLogging hidden label to Advanced tab
- Add FunctionalButtonShowLog identifier to Show log viewer button on
  Advanced tab
- Add FunctionalButtonLogDirectory identifier to Open log directory
  button on Advanced tab
- Add FunctionalButtonShowConfig identifier to Show the config file in
  Finder button on Key Bindings tab
- Add FunctionalButtonImportConfig identifier to Import an existing
  config file button on Key Bindings tab
- Add FunctionalButtonDefaultApp identifier to Set IINA as the Default
  Application button on Utilities tab
- Add formSearchTerm method to PreferenceWindowController that removes
  extraneous characters from labels and titles
- Correct the spelling of available on General tab
- Change Use media keys to be  Use system media control to match UI
- Add a logLabelDict method to PreferenceWindowController that logs
  the search terms found by scanning NIBs

low-batt commented:
**NOTE:** To reviewers, have a look at my post in the issue (#4512). That explains the changes and shows the search terms that are now being extracted from the settings panel NIBs.

My question for reviewers is whether I am being too aggressive in removing punctuation from the search terms. In other words should searching for "Layout:", with the ":" at the end return results? The changed code is stripping the colon, so now it will not be found. Of course as you are typing "Layout", unless you are a fast typist you will get results before you can type the colon.

Compare the debug output in the issue to see the changes.

uiryuu commented on iina/en.lproj/PrefGeneralViewController.strings:
```diff
@@ -123,7 +123,7 @@
 "jeK-46-2BG.title" = "History:";
 
 /* Class = "NSButtonCell"; title = "Show artist and track name for audio files when avaliable"; ObjectID = "kBI-Th-bBV"; */
```
Please also correct it in the comment for consistency purposes

low-batt replied:
Fixed

uiryuu commented on iina/en.lproj/PrefKeyBindingViewController.strings:
```diff
@@ -1,5 +1,5 @@
 /* Class = "NSButtonCell"; title = "Use media keys"; ObjectID = "2WY-CR-baD"; */
```
Ditto

low-batt replied:
Fixed

uiryuu reviewed (changes requested):
I noticed that the plugins settings are not searchable as well. Is it intentional for now?

low-batt committed:
Fix the Advanced Settings Aren't Searchable, #4512

This commit will:
- Add SectionTitleTop hidden label to Advanced tab
- Add SectionTitleLogging hidden label to Advanced tab
- Add FunctionalButtonEnableAdvanced identifier to Enable advanced
  settings button on Advanced tab
- Add FunctionalButtonShowLog identifier to Show log viewer button on
  Advanced tab
- Add FunctionalButtonLogDirectory identifier to Open log directory
  button on Advanced tab
- Add FunctionalButtonShowConfig identifier to Show the config file in
  Finder button on Key Bindings tab
- Add FunctionalButtonImportConfig identifier to Import an existing
  config file button on Key Bindings tab
- Add FunctionalButtonDefaultApp identifier to Set IINA as the Default
  Application button on Utilities tab
- Add SectionTitlePlugins hidden label to Plugins tab
- Add FunctionalButtonInstallGitHub identifier to Install from GitHub
  button on Plugins tab
- Add FunctionalButtonInstallLocal identifier to Install from a local
  package button on Plugins tab
- Add FunctionalButtonShowPlugin identifier to Show in Finder button on
  Plugins tab
- Add formSearchTerm method to PreferenceWindowController that removes
  extraneous characters from labels and titles
- Correct the spelling of available on General tab
- Change Use media keys to be Use system media control to match UI
- Add a logLabelDict method to PreferenceWindowController that logs
  the search terms found by scanning NIBs

low-batt commented:
I was avoiding the plugins settings over worries about merge conflicts. I have added them in this new commit. These are the terms from the plugins settings:
```text
20:02:37.339 [iina][d] Plugins
20:02:37.339 [iina][d]   Plugins
20:02:37.339 [iina][d]     Uninstall
20:02:37.339 [iina][d]     Show in Finder
20:02:37.339 [iina][d]     Installed plugins
20:02:37.339 [iina][d]     Install from GitHub...
20:02:37.339 [iina][d]     Plugins
20:02:37.339 [iina][d]     Install from a local package
```

low-batt committed:
Fix the Advanced Settings Aren't Searchable, #4512

This commit will:
- Add SectionTitleTop hidden label to Advanced tab
- Add SectionTitleLogging hidden label to Advanced tab
- Add FunctionalButtonEnableAdvanced identifier to Enable advanced
  settings button on Advanced tab
- Add FunctionalButtonShowLog identifier to Show log viewer button on
  Advanced tab
- Add FunctionalButtonLogDirectory identifier to Open log directory
  button on Advanced tab
- Add FunctionalButtonShowConfig identifier to Show the config file in
  Finder button on Key Bindings tab
- Add FunctionalButtonImportConfig identifier to Import an existing
  config file button on Key Bindings tab
- Add FunctionalButtonDefaultApp identifier to Set IINA as the Default
  Application button on Utilities tab
- Add SectionTitlePlugins hidden label to Plugins tab
- Add FunctionalButtonInstallGitHub identifier to Install from GitHub
  button on Plugins tab
- Add FunctionalButtonInstallLocal identifier to Install from a local
  package button on Plugins tab
- Add FunctionalButtonShowPlugin identifier to Show in Finder button on
  Plugins tab
- Add formSearchTerm method to PreferenceWindowController that removes
  extraneous characters from labels and titles
- Correct the spelling of available on General tab
- Change Use media keys to be Use system media control to match UI
- Add a logLabelDict method to PreferenceWindowController that logs
  the search terms found by scanning NIBs
- Add a trimWhitespaceSuffix method to the private String extension
- Correct an existing defect in the removedLastSemicolon method
- Change searchFieldAction to remove trailing whitespace and a
  trailing colon if present from the string to search for

low-batt commented:
After discussion with @uiryuu I have added stripping of any trailing whitespace or colon from the user's search string.
